src/lib.rs

```diff
@@ -152,10 +200,8 @@ impl Watch {
     }
 
     pub fn exclude_workspace_path(&mut self, path: impl AsRef<Path>) {
-        let metadata = metadata();
-
         self.workspace_exclude_paths
-            .push(metadata.workspace_root.as_std_path().join(path))
+            .push(metadata().workspace_root.as_std_path().join(path))
```

Comment:
The fact that you transform the relative paths to absolute paths in this method renders the field workspace_exclude_pathscompletely useless. You could simply add to the vector exclude_paths.

The reason why I wanted 2 different vectors in the first place is because it would allow the user to search in the vector for a relative path relative to the workspace. This is now impossible because you force it to absolute paths.